rsmtool/utils/prmse.py

```diff
@@ -0,0 +1,328 @@
+
+    # compute mean human score and total number of scores
+    # for each response
+    mean_scores = np.nanmean(human_scores, axis=1)
+    n_scores = get_n_human_scores(human_scores)
+
+    # compute overall mean
+    mean_human_score = np.nanmean(human_scores)
+
+    # let N be total number of responses
+    N = len(human_scores)
+
+    # let M be total number of human ratings
+    M = n_scores.sum()
+
+    # compute squared deviations
+    squared_devs = (mean_scores - mean_human_score)**2
+
+    adjusted_squared_devs = n_scores * squared_devs
+
+    sum_of_squares = adjusted_squared_devs.sum()
+
+    numerator = sum_of_squares - (N-1) * variance_errors_human
+
+    denominator = M - ((n_scores**2).sum() / M)
+
+    variance_true_scores = numerator / denominator
+
+    return variance_true_scores
+
+
+def mse_true(system,
+             human_scores,
+             variance_errors_human=None):
+
+    """
+    Compute mean square error (MSE) when predicting true score
+    from system score.
+
+    Parameters
+    ----------
+    system : array-like of shape (n_samples,)
+        System scores
+    human_scores : array-like of shape (n_samples, n_ratings)
+        Human ratings for each response
+    variance_errors_human : float
+        Estimated variance of errors in human scores
+        When no value is supplied, the variance will
+        be estimated from the data. In this case
+        at least some responses must have more than
+        one human score.
+
+
+    Returns
+    -------
+    variance_true_scores : float
+        Variance of true scores
+    """
+
+    # if we don't have variance of errors, compute it
+    # from the data
+
+    if not variance_errors_human:
+        variance_errors_human = variance_of_errors(human_scores)
+
+
+    # get total number of scores for each response
+    n_scores = get_n_human_scores(human_scores)
+    mean_scores = np.nanmean(human_scores, axis=1)
+
+    N = len(system)
+
+    se = ((mean_scores - system)**2) * n_scores
+
+    # Compute mean squared error when predicting true score
+    mse = (se.sum() - N * variance_errors_human) / n_scores.sum()
+    return mse
+
+
+def prmse_true(system,
+               human_scores,
+               variance_errors_human=None):
+    """
+    Compute Proportional Reduction in Mean Squared Error (PRMSE)
+    when predicting true score from system scores.
+
+    Parameters
+    ----------
+    system : array-like of shape (n_samples,)
+        System scores
+    human_scores : array-like of shape (n_samples, n_ratings)
+        Human ratings for each response
+    variance_errors_human : float
+        Estimated variance of errors in human scores
+        When no value is supplied, the variance will
+        be estimated from the data. In this case
+        at least some responses must have more than
+        one human score.
+
+    Returns
+    -------
+    prmse : float
+        Proportional reduction in mean square error
```

Comment:
Is it "mean squared error" or "mean square error"? I had the same question for `mse_true()` as well.